int x;
int y;
int weight;
int observed;